scripts/consume_flow_with_threads_test.py

```diff
@@ -0,0 +1,216 @@
+            bytess=[b""],
+        )
+        ddo = ocean.assets.create(
+            metadata=metadata,
+            publisher_wallet=wallet,
+            encrypted_files=encrypted_files,
+            erc721_address=erc721_nft.address,
+            erc20_tokens_data=[erc20_data],
+        )
+        assert ddo, "The asset is not created."
+        assert ddo.nft["name"] == "NFT"
+        assert ddo.nft["symbol"] == "NFTSYMBOL"
+        assert ddo.nft["address"] == erc721_nft.address
+        assert ddo.nft["owner"] == wallet.address
+        assert ddo.datatokens[0]["name"] == "Datatoken 1"
+        assert ddo.datatokens[0]["symbol"] == "DT1"
+        assert ddo.credentials == build_credentials_dict()
+
+        service = ddo.get_service(ServiceTypes.ASSET_ACCESS)
+
+        # Initialize service
+        response = data_provider.initialize(
+            did=ddo.did, service=service, consumer_address=wallet.address
+        )
+        assert response
+        assert response.status_code == 200
+        assert response.json()["providerFee"]
+
+        # Consume fees
+        consume_fees = ConsumeFees(
+            consumer_market_fee_address=wallet.address,
+            consumer_market_fee_token=erc20_token.address,
+            consumer_market_fee_amount=0,
+        )
+
+        erc20_token.mint(
+            account_address=wallet.address,
+            value=ocean.to_wei(i + 1),
+            from_wallet=wallet,
+        )
+        # Start order for consumer
+        tx_id = erc20_token.start_order(
+            consumer=wallet.address,
+            service_index=ddo.get_index_of_service(service),
+            provider_fees=response.json()["providerFee"],
+            consume_fees=consume_fees,
+            from_wallet=wallet,
+        )
+        # Download file
+        destination = config.downloads_path
+        if not os.path.isabs(destination):
+            destination = os.path.abspath(destination)
+
+        if os.path.exists(destination) and len(os.listdir(destination)) > 0:
+            list(
+                map(
+                    lambda d: shutil.rmtree(os.path.join(destination, d)),
+                    os.listdir(destination),
+                )
+            )
+
+        if not os.path.exists(destination):
+            os.mkdir(destination)
+
+        assert len(os.listdir(destination)) == 0
+
+        ocean.assets.download_asset(
+            asset=ddo,
+            consumer_wallet=wallet,
+            destination=destination,
+            order_tx_id=tx_id,
+        )
+
+        assert (
+            len(os.listdir(os.path.join(destination, os.listdir(destination)[0]))) > 0
+        ), "The asset folder is empty."
+
+
+def thread_function1(ocean, alice_wallet, config):
+    consume_flow(ocean, wallet=alice_wallet, config=config)
+
+
+def thread_function2(ocean, bob_wallet, config):
+    consume_flow(ocean, wallet=bob_wallet, config=config)
+
+
+def thread_function3(ocean, tristan_wallet, config):
```

Comment:
The same advice I gave for the compute flow stress tests applies here:

1. The consume flow should be placed in a reusable function to eliminate duplication.
1. Using `ThreadPoolExecutor` would cut down on the `threading.Thread` boiler plate.
3. I'd like to see a test with more concurrency and less duration.